workflow/workflow_test.go

```diff
@@ -306,6 +306,111 @@ func TestWithDoubleSerializationDetection(t *testing.T) {
 	// Confirm that, while we did no double-unserializations or double-serializations,
 	// we did do at least one single one.
 	assert.Equals(t, errorDetect.SerializeCnt+errorDetect.UnserializeCnt > 0, true)
+
+	// again, but with a default value for the error detector
+	errorDetect = util.NewInvalidSerializationDetectorSchema()
+	// Inject the error detector into the object
+	rootObject.PropertiesValue["error_detector"] = schema.NewPropertySchema(
+		errorDetect,
+		nil,
+		true,
+		nil,
+		nil,
+		nil,
+		schema.PointerTo[string]("default"),
+		nil,
+	)
+	outputID, _, err = preparedWorkflow.Execute(context.Background(), map[string]any{
+		//"error_detector": "original input",
+	})
+	assert.NoError(t, err)
+	assert.Equals(t, outputID, "a")
+	// Confirm that, while we did no double-unserializations or double-serializations,
+	// we did do at least one single one.
+	assert.Equals(t, errorDetect.SerializeCnt+errorDetect.UnserializeCnt > 0, true)
```

Comment:
Rather than cookie-cuttering this code, I would include it only once and drive it with a table.  E.g.,
```go
	type testIterType struct {
		defaultSpec	*string
		input		map[string]any
	}
	testIter = []testIterType{
		// No default specified; input provided
		testIterType{
			nil,
			map[string]any{"error_detector": "original input"},
		},
		// Default specified; input provided (overrides default)
		testIterType{
			schema.PointerTo[string]("default"),
			map[string]any{"error_detector": "original input"},
		},
		// Default specified; input omitted (default value used)
		testIterType{
			schema.PointerTo[string]("default"),
			map[string]any{},
		},
	}

	for _, i := range testIter {
		errorDetect := util.NewInvalidSerializationDetectorSchema()
		// Inject the error detector into the object
		rootObject.PropertiesValue["error_detector"] = schema.NewPropertySchema(
			errorDetect,
			nil,
			true,
			nil,
			nil,
			nil,
			i.defaultSpec,
			nil,
		)
		outputID, _, err := preparedWorkflow.Execute(context.Background(), i.input)
		assert.NoError(t, err)
		assert.Equals(t, outputID, "a")
		// Confirm that, while we did no double-unserializations or double-serializations,
		// we did do at least one single one.
		assert.Equals(t, errorDetect.SerializeCnt+errorDetect.UnserializeCnt > 0, true)
	}
```